encoder/csv.go

```diff
@@ -200,7 +170,34 @@ func (c *CSVItemIterator) Close() error {
 
 func (c *CSVItemIterator) Read() (common_datalayer.Item, error) {
 	// TODO: fix me
-	return nil, nil
+	record, err := c.decoder.Read()
+
+	// create item from record
```

Comment:
Please tidy up comments, remove commented out code if we arent going to have it.